<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>frmFlatnessClass</class>
 <widget class="QDialog" name="frmFlatnessClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>450</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>450</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>拟合平面/计算点到平面距离</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(220, 220, 220);</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>1</number>
   </property>
   <item row="2" column="0">
    <spacer name="verticalSpacer_4">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>5</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>42</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="3" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>6</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>10</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>83</width>
         <height>25</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgba(255, 255, 255, 0);</string>
       </property>
       <property name="text">
        <string>单元描述：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtDescribe">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>357</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>30</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_8">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnExecute">
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>37</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QWidget{ background-color: rgb(85, 135, 127);}
QWidget{ color: rgb(235, 235, 235);}
QWidget{ border: 2px groove gray;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgb(85, 155, 127);}
QPushButton:pressed{background-color: rgb(85, 175, 127);}</string>
       </property>
       <property name="text">
        <string>执行</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>6</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QTabWidget" name="tabWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>450</width>
       <height>350</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>微软雅黑</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #bcc2bc;</string>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>基本设置</string>
      </attribute>
      <widget class="QGroupBox" name="groupCamera">
       <property name="geometry">
        <rect>
         <x>9</x>
         <y>14</y>
         <width>439</width>
         <height>185</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #dfd5cf, stop: 0.15 #acbbac, stop: 0.85 #acbbac,stop:1 #dfd5cf);
color: rgb(85, 0, 255);</string>
       </property>
       <property name="title">
        <string>拟合平面</string>
       </property>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>42</y>
          <width>91</width>
          <height>19</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(30, 30, 30);
background: transparent;</string>
        </property>
        <property name="text">
         <string>输入点集X：</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="txtLinkPointsX">
        <property name="geometry">
         <rect>
          <x>106</x>
          <y>39</y>
          <width>255</width>
          <height>27</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>全局变量double[]类型</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 200);
background-color: rgba(170, 170, 127,50);</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="btnLinkPointsX">
        <property name="geometry">
         <rect>
          <x>366</x>
          <y>39</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>链接输入点集X</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(85, 150, 127, 100);}
QPushButton:pressed{background-color: rgba(85, 180, 127, 160);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/link.png</normaloff>:/resource/link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QPushButton" name="btnDelLinkPointsX">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>39</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>删除输入点集X链接</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(226, 142, 162, 100);}
QPushButton:pressed{background-color: rgba(226, 172, 162, 200);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/del_link.png</normaloff>:/resource/del_link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QPushButton" name="btnLinkPointsY">
        <property name="geometry">
         <rect>
          <x>366</x>
          <y>92</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>链接输入点集Y</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(85, 150, 127, 100);}
QPushButton:pressed{background-color: rgba(85, 180, 127, 160);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/link.png</normaloff>:/resource/link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QPushButton" name="btnDelLinkPointsY">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>92</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>删除输入点集Y链接</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(226, 142, 162, 100);}
QPushButton:pressed{background-color: rgba(226, 172, 162, 200);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/del_link.png</normaloff>:/resource/del_link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QLineEdit" name="txtLinkPointsY">
        <property name="geometry">
         <rect>
          <x>106</x>
          <y>92</y>
          <width>255</width>
          <height>27</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>全局变量double[]类型</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 200);
background-color: rgba(170, 170, 127,50);</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>95</y>
          <width>91</width>
          <height>19</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(30, 30, 30);
background: transparent;</string>
        </property>
        <property name="text">
         <string>输入点集Y：</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>151</y>
          <width>91</width>
          <height>19</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(30, 30, 30);
background: transparent;</string>
        </property>
        <property name="text">
         <string>输入点集Z：</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="txtLinkPointsZ">
        <property name="geometry">
         <rect>
          <x>106</x>
          <y>148</y>
          <width>255</width>
          <height>27</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>全局变量double[]类型</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 200);
background-color: rgba(170, 170, 127,50);</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="btnDelLinkPointsZ">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>148</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>删除输入点集Z链接</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(226, 142, 162, 100);}
QPushButton:pressed{background-color: rgba(226, 172, 162, 200);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/del_link.png</normaloff>:/resource/del_link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QPushButton" name="btnLinkPointsZ">
        <property name="geometry">
         <rect>
          <x>366</x>
          <y>148</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>链接输入点集Z</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(85, 150, 127, 100);}
QPushButton:pressed{background-color: rgba(85, 180, 127, 160);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/link.png</normaloff>:/resource/link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupCamera_2">
       <property name="geometry">
        <rect>
         <x>9</x>
         <y>210</y>
         <width>439</width>
         <height>100</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #dfd5cf, stop: 0.15 #acbbac, stop: 0.85 #acbbac,stop:1 #dfd5cf);
color: rgb(85, 0, 255);</string>
       </property>
       <property name="title">
        <string>计算点到平面距离</string>
       </property>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>67</y>
          <width>91</width>
          <height>19</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(30, 30, 30);
background: transparent;</string>
        </property>
        <property name="text">
         <string>三维输入点：</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="txtLink3DPoint">
        <property name="geometry">
         <rect>
          <x>106</x>
          <y>64</y>
          <width>255</width>
          <height>27</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>全局变量cv::Point3d类型</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 200);
background-color: rgba(170, 170, 127,50);</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="btnLink3DPoint">
        <property name="geometry">
         <rect>
          <x>366</x>
          <y>64</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>链接三维输入点</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(85, 150, 127, 100);}
QPushButton:pressed{background-color: rgba(85, 180, 127, 160);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/link.png</normaloff>:/resource/link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QPushButton" name="btnDelLink3DPoint">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>64</y>
          <width>27</width>
          <height>27</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>删除三维输入点链接</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QWidget{ border:none;border-radius: 8px;padding: 2px 4px;}
QPushButton:hover{ background-color: rgba(226, 142, 162, 100);}
QPushButton:pressed{background-color: rgba(226, 172, 162, 200);}</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="sub_res.qrc">
          <normaloff>:/resource/del_link.png</normaloff>:/resource/del_link.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>27</width>
          <height>27</height>
         </size>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkDist">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>28</y>
          <width>169</width>
          <height>27</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QCheckBox{
    color: rgb(30, 30, 30);
	border-style: flat;
	background: transparent;
}
QCheckBox::indicator {
    width: 17px;
    height: 17px;    
}
QCheckBox::indicator::unchecked {
    image: url(:/resource/untick.png);
}
QCheckBox::indicator::checked {
    image: url(:/resource/tick.png);
};</string>
        </property>
        <property name="text">
         <string>计算点到平面距离</string>
        </property>
        <property name="iconSize">
         <size>
          <width>22</width>
          <height>22</height>
         </size>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupC3">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>14</y>
         <width>325</width>
         <height>296</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #dfd5cf, stop: 0.15 #acbbac, stop: 0.85 #acbbac,stop:1 #dfd5cf);
color: rgb(85, 0, 255);</string>
       </property>
       <property name="title">
        <string>输出数据(A、B、C、D、Distance)</string>
       </property>
       <widget class="QTextEdit" name="txtMsg">
        <property name="geometry">
         <rect>
          <x>4</x>
          <y>32</y>
          <width>317</width>
          <height>259</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑</family>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QTextEdit         
{
    background-color: rgba(180, 180, 215,120);    
    gridline-color: rgba(220, 220, 255,255);
    color: rgba(20, 20, 20,200);
}
QScrollBar:vertical{ 
		background:#FFFFF2;
		padding-top:20px;
		padding-bottom:20px;
		padding-left:2px;
		padding-right:2px;
		border-left:1px solid #d7d7d7;}
QScrollBar::handle:vertical{
		background:#dbdbdb;
		border-radius:6px;
		min-height:20px;}
QScrollBar::handle:vertical:hover{
		background:#d0d0d0;}
QScrollBar::add-line:vertical{
		background:url(:/resource/down.png) center no-repeat;}
QScrollBar::sub-line:vertical{
		background:url(:/resource/up.png) center no-repeat;}
QScrollBar:horizontal{ 
		background:#FFFFF2;
		padding-top:20px;
		padding-bottom:20px;
		padding-left:2px;
		padding-right:2px;
		border-left:1px solid #d7d7d7;}
QScrollBar::handle:horizontal{
		background:#dbdbdb;
		border-radius:6px;
		min-height:20px;}
QScrollBar::handle:horizontal:hover{
		background:#d0d0d0;}
QScrollBar::add-line:horizontal{
		background:url(:/resource/right.png) center no-repeat;}
QScrollBar::sub-line:horizontal{
		background:url(:/resource/left.png) center no-repeat;}</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="sub_res.qrc"/>
  <include location="../../ImageProcess/ImageFlip/sub_res.qrc"/>
 </resources>
 <connections/>
</ui>
